src/base_maths.jl

```diff
@@ -162,11 +162,11 @@ function Base.:*(M::Diagonal{T}, B::BlockDiagonal{T2})::BlockDiagonal where {T,
     return A
 end
 
-function Base.:*(vt::Adjoint{T,<: AbstractVector}, B::BlockDiagonal{T}) where {T}
+function Base.:*(vt::Adjoint{T,<: AbstractVector}, B::BlockDiagonal{T2, V}) where V<:AbstractMatrix{T2} where {T, T2}
```

Comment:
This is typically written as
```suggestion
function Base.:*(vt::Adjoint{<:Any,<:AbstractVector}, B::BlockDiagonal)
```
or you could additionally import `LinearAlgebra.AdjointAbsVec`. I think a general rule is to introduce such parameters only when you need them in the function body. Otherwise just skip all parameters, or fill the leading ones with `<:Any`.

Reply:
Thanks - this is much nicer!